ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+                        return 1;
+                }
+            }
+            else if (pBD_XP->pField[nIField + nNumPrivateMMField].FieldType ==
+                     'D')
+            {
+                if (MMWriteValueToRecordDBXP(hMiraMonLayer, pszRecordOnCourse,
+                                             pBD_XP->pField + nIField +
+                                                 nNumPrivateMMField,
+                                             hMMFeature->pRecords[nIRecord]
+                                                 .pField[nIField]
+                                                 .pDinValue,
+                                             FALSE))
+                    return 1;
+            }
+        }
+
+        if (MMAppendBlockToBuffer(pFlushRecList))
+            return 1;
+
+        (*nNumRecords)++;
+    }
+    return 0;
+}
+
+// Adds feature records to a MiraMon database associated with a vector layer.
+static int MMDetectAndFixDBFWidthChange(
+    struct MiraMonVectLayerInfo *hMiraMonLayer,
+    struct MiraMonFeature *hMMFeature, struct MMAdmDatabase *pMMAdmDB,
+    struct MM_FLUSH_INFO *pFlushRecList, MM_EXT_DBF_N_FIELDS nNumPrivateMMField,
+    MM_EXT_DBF_N_MULTIPLE_RECORDS nIRecord, MM_EXT_DBF_N_FIELDS nIField)
+{
+    if (!hMiraMonLayer)
+        return 1;
+
+    if (!hMMFeature)
+        return 0;
+
+    if (nIRecord >= hMMFeature->nNumMRecords)
+        return 0;
+
+    if (nIField >= hMMFeature->pRecords[nIRecord].nNumField)
+        return 0;
+
+    if (MMTestAndFixValueToRecordDBXP(
+            hMiraMonLayer, pMMAdmDB, nIField + nNumPrivateMMField,
+            hMMFeature->pRecords[nIRecord].pField[nIField].pDinValue))
+        return 1;
+
+    // We analize next fields
+    if (nIField == hMMFeature->pRecords[nIRecord].nNumField - 1)
+    {
+        if (MMDetectAndFixDBFWidthChange(hMiraMonLayer, hMMFeature, pMMAdmDB,
+                                         pFlushRecList, nNumPrivateMMField,
+                                         nIRecord + 1, 0))
+            return 1;
+    }
+    else
+    {
+        if (MMDetectAndFixDBFWidthChange(hMiraMonLayer, hMMFeature, pMMAdmDB,
+                                         pFlushRecList, nNumPrivateMMField,
+                                         nIRecord, nIField + 1))
+            return 1;
+    }
+    return 0;
+}  // End of MMDetectAndFixDBFWidthChange()
+
+// Adds a DBF record to a MiraMon table associated with a vector layer.
+// It sets up flush settings for writing to the table and initializes
+// variables needed for the process. Then, it checks and fixes the width
+// change if necessary.
+int MMAddDBFRecordToMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                         struct MiraMonFeature *hMMFeature)
+{
+    struct MM_DATA_BASE_XP *pBD_XP = nullptr;
+    MM_EXT_DBF_N_FIELDS nNumPrivateMMField = 0;
+    char *pszRecordOnCourse;
+    struct MM_FLUSH_INFO *pFlushRecList;
+
+    if (!hMiraMonLayer)
+        return 1;
```

Comment:
other code paths in that function use symbolic names, so
```suggestion
        return MM_FATAL_ERROR_WRITING_FEATURES;
```

Reply:
Here and other places. Done